gio/src/file.rs

```diff
@@ -1111,6 +1113,137 @@ pub trait FileExtManual: IsA<File> + Sized {
 
         (fut, Box::pin(receiver))
     }
+
+    #[doc(alias = "g_file_set_attribute")]
+    fn set_attribute(
+        &self,
+        attribute: &str,
+        type_value: FileAttributeTypeValue,
+        flags: FileQueryInfoFlags,
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = std::ptr::null_mut();
+            let is_ok = ffi::g_file_set_attribute(
+                self.as_ref().to_glib_none().0,
+                attribute.to_glib_none().0,
+                type_value.type_().into_glib(),
+                type_value.to_glib_none().0,
+                flags.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+// checker-ignore-item
+pub enum FileAttributeTypeValue<'a> {
```

Comment:
I think I would gone with a trait but this also works. Should probably have `From<T>` impls to easily convert from plain types to this

Reply:
Could then also use `impl Into<FileAttributeTypeValue>` in the function.

Also I wonder if the same type could be used for the other file attribute API

Reply:
Not very different from `object.set_property()` and other such API though